pkg/virt-controller/watch/vmi.go

```diff
@@ -307,26 +309,58 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 		}
 	}
 
+	hasWffcDataVolume := false
+	// there is no reason to check for waitForFirstConsumer is there are failed DV's
+	if !hasFailedDataVolume {
+		for _, dataVolume := range dataVolumes {
+			if dataVolume.Status.Phase == cdiv1.WaitForFirstConsumer {
+				hasWffcDataVolume = true
+			}
+		}
+	}
+
 	conditionManager := controller.NewVirtualMachineInstanceConditionManager()
 	vmiCopy := vmi.DeepCopy()
-	podExists := podExists(pod)
+	vmiPodExists := podExists(pod) && !isTempPod(pod)
+	tempPodExists := podExists(pod) && isTempPod(pod)
 
 	vmiCopy, err := c.setActivePods(vmiCopy)
 	if err != nil {
 		return fmt.Errorf("Error detecting vmi pods: %v", err)
 	}
 
 	switch {
-
 	case vmi.IsUnprocessed():
-		if podExists {
+		if vmiPodExists {
 			vmiCopy.Status.Phase = virtv1.Scheduling
 		} else if vmi.DeletionTimestamp != nil {
 			vmiCopy.Status.Phase = virtv1.Failed
 		} else if hasFailedDataVolume {
 			vmiCopy.Status.Phase = virtv1.Failed
 		} else {
 			vmiCopy.Status.Phase = virtv1.Pending
+			if hasWffcDataVolume {
+				if !tempPodExists {
+					condition := virtv1.VirtualMachineInstanceCondition{
+						Type:   virtv1.VirtualMachineInstanceProvisioning,
+						Status: k8sv1.ConditionTrue,
+					}
+					if !conditionManager.HasCondition(vmiCopy, condition.Type) {
+						vmiCopy.Status.Conditions = append(vmiCopy.Status.Conditions, condition)
+					}
+				} else {
+					// Add PodScheduled False condition to the VM
+					if cond := conditionManager.GetPodConditionWithStatus(pod, k8sv1.PodScheduled, k8sv1.ConditionFalse); cond != nil {
+						conditionManager.AddPodCondition(vmiCopy, cond)
+					} else if conditionManager.HasCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodScheduled)) {
+						// Remove PodScheduling condition from the VM
+						conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodScheduled))
+					}
+					if isPodDownOrGoingDown(pod) || isPodSchedulingError(pod) {
+						vmiCopy.Status.Phase = virtv1.Failed
```

Comment:
We should maintain the same behavior for this temporary pod as we have for the normal pod that actually hosts the VMI. The PodReasonUnschedulable is a transient condition. It can clear up if more resources are added to the cluster

```
        // PodReasonUnschedulable reason in PodScheduled PodCondition means that the scheduler
        // can't schedule the pod right now, for example due to insufficient resources in the cluster.
```

Right now, VMIs don't fail if this condition is encountered with the pod that hosts the qemu process. The behavior for the temporary pod shoudl be the same. 

To maintain the same behavior as it is today, we'd just fail the VMI if the temp pod is going down or is down.

Reply:
Ok if that is the case, then I am updating it - removing the check for PodReasonUnschedulable.